src/components/Search/Search.js

```diff
@@ -0,0 +1,33 @@
+import React, { useState } from 'react';
+
+function Search({ initialQuery, onSearch }) {
+  const [query, setQuery] = useState(initialQuery);
+
+  const handleInputChange = (e) => {
+    setQuery(e.target.value);
+  };
+
+  const handleKeyPress = (e) => {
+    if (e.key === 'Enter') {
+      onSearch(query);
+    }
+  };
+
+  const handleClick = () => {
+    onSearch(query);
+  };
+
+  return (
+    <div>
```

Comment:
Inbox use of the **HTML Form submit** event to handle both cases.
Use `<form onSubmit={handleSubmit}>` so we can use `<button type="submit">Search</button>` to handle the `Enter` key with button click in one function:

```jsx
  const handleSubmit = (e) => {
    e.preventDefault();
    props.onSubmit(query);
  };

  return (
    <form onSubmit={handleSubmit}>
      <input
        type="text"
        value={query}
        onChange={handleInputChange}
      />
      <button type="submit">Search</button>
    </form>
```